doc/design.rst

```diff
@@ -0,0 +1,113 @@
+#####################
+Design with use-cases
+#####################
+
+We like to write tutorials and other documents in restructured text, and
+particularly with sphinx.
+
+This allows us to use - say - vim - for editing text and code.
+
+The IPython notebook is pleasantly interactive in showing us rendering of LaTex,
+and output of code execution, but the web interface feels heavy and
+uncomfortable when writing anything longer than a few pages.
+
+*********
+Use-cases
+*********
+
+* I write a short snippet of code in a RST file in vim.  From vim I cause the
+  code to be run, and the results are shown in the web browser, including
+  generated plots.  I can cause the printed results from the notebook to be
+  output into my current text buffer in vim;
+* When building the final page from the RST document, the rendered plots from
+  the web browser will appear as images in the rendered page;
+* From vim I can cause some current part of text to be converted to markdown and
+  rendered in the web browser.  This allows me to review LaTeX rendered with
+  MathJax or similar;
+* As I edit, I may want to keep a selection of rendered results current in the
+  web browswer to remind me what the rendered output will look like;
+* I will often want to check the results of the code in the RST file still hold
+  after my edits.  I can run some command from vim to rerun all the code
+  snippets in the vim buffer in a fresh execution environment and replace the
+  rendered results in the web browser.
+
+*********************
+Implementation sketch
+*********************
+
+We edit text on the (say) left of the screen, in vim, and review output on the
+(say) right of the screen, in some version of the IPython notebook.
+
+This is *edit left, look right*.
+
+We stored the results that are the basis for the display on the right, as an
+IPython notebook, in notebook JSON format.  We use this notebook as store for
+the results, and we will fetch the results from this store when creating the
```

Comment:
`will fetch the assets from this store` - since it's not just results (`Out` prompts), but also display results (mime-typed stdout/stderrr)